specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+      ...lineData,
+      fieldsData: Object.fromEntries(
+        Object.entries(lineData.fieldsData).map(([field, data]) => [
+          field,
+          { ...data, isDefault: mappingPath[index] === field },
+        ])
+      ),
+    }));
+
+  const databaseLineData: RA<MappingLineData> = React.useMemo(
+    () => [
+      {
+        customSelectSubtype: 'simple',
+        tableName: model.name,
+        fieldsData: {
+          ...databaseScopeData,
+          ...getValidScopeRelationships(model),
+        },
+      },
+    ],
+    [model]
+  );
+
+  const [lineData, setLineData] = React.useState(
+    updateLineData(
+      mappingPath.map((_, index) => {
+        const databaseScope = index === 0 ? databaseScopeData : {};
+        const modelPath =
+          index === 0
+            ? model
+            : getFieldsFromPath(model, rule.scopes[0])[index].model;
+        return {
+          customSelectSubtype: 'simple',
+          tableName: modelPath.name,
+          fieldsData: {
+            ...databaseScope,
+            ...getValidScopeRelationships(modelPath),
+          },
+        };
+      }),
+      mappingPath
+    )
+  );
+
+  const getRelationshipData = (newTableName: keyof Tables): MappingLineData => {
+    const newModel = strictGetModel(newTableName);
+
+    return {
+      customSelectSubtype: 'simple',
+      tableName: newModel.name,
+      fieldsData: getValidScopeRelationships(newModel),
+    };
+  };
+
+  return (
+    <MappingView
+      mappingElementProps={getMappingLineProps({
+        mappingLineData: lineData,
+        customSelectType: 'OPENED_LIST',
+        onChange({ isDoubleClick, isRelationship, index, ...rest }) {
+          if (isRelationship) {
+            const newMappingPath = replaceItem(
+              mappingPath.slice(0, index + 1),
+              index,
+              rest.newValue
+            );
+            setMappingPath(newMappingPath);
+            setLineData((lineData) =>
+              updateLineData(
+                [
+                  ...lineData.slice(0, index + 1),
+                  getRelationshipData(rest.newTableName!),
+                ],
+                newMappingPath
+              )
+            );
+            if (isDoubleClick)
+              handleChanged({ ...rule, scopes: [mappingPath.join('__')] });
+          } else {
+            setMappingPath(['database']);
+            setLineData(databaseLineData);
+            if (isDoubleClick)
+              handleChanged({
+                ...rule,
+                scopes: [],
+              });
+          }
+        },
+      })}
+    >
+      <Button.Small
+        aria-label={schemaText.setScope()}
+        className="justify-center p-2"
+        title={schemaText.setScope()}
+        onClick={(): void => {
+          handleChanged({
+            ...rule,
+            scopes:
+              mappingPath.length === 1 && mappingPath[0] === 'database'
+                ? []
+                : [mappingPath.join('__')],
```

Comment:
this is bug-prone. extract this "magic" string into a variable

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/43aee5b97e52edca55b58e0cb2b76773754b510f